include/aikido/planner/World.hpp

```diff
@@ -39,6 +39,11 @@ class World
   /// \param name Name of desired Skeleton
   dart::dynamics::SkeletonPtr getSkeleton(const std::string& name) const;
 
+  /// Find a Skeleton by Skeleton, returning null if it is not in the World
+  /// \param skel Desired Skeleton
+  dart::dynamics::SkeletonPtr getSkeleton(
+      const dart::dynamics::SkeletonPtr skel) const;
```

Comment:
Nit: Let's use `const dart::dynamics::SkeletonPtr&` to avoid unnecessary increase of the reference count by copying `std::shared_ptr<...>`.

Reply:
I don't think we can use `std::weak_ptr` here because it is not comparable or hashable. This makes sense because an unlocked `weak_ptr` can spontaneously degrade to `nullptr` if all owning `shared_ptr`s are destructed.

We could using a `const Skeleton*` as the key. But we'd need to be careful to not access the pointer after the `Skeleton` is destructed, e.g. by `lock()`ing a `std::weak_ptr` that we store inside the marker instance.